state-chain/chains/src/sol/instruction_builder.rs

```diff
@@ -121,9 +130,28 @@ impl SolanaInstructionBuilder {
 						SolAccountMeta::new_readonly(agg_key.into(), true),
 						SolAccountMeta::new_readonly(fetch_id.address.into(), false),
 						SolAccountMeta::new(ata.address.into(), false),
-						SolAccountMeta::new(token_vault_ata.into(), false),
-						SolAccountMeta::new_readonly(token_mint_pubkey.into(), false),
-						SolAccountMeta::new_readonly(token_program_id.into(), false),
+						SolAccountMeta::new(
+							token_environments
+								.get(&SolAsset::SolUsdc)
```

Comment:
This is also still usdc-specific. Shouldn't we update `AssetWithDerivedAddress` to have a token variant? `AssetWithDerivedAddress::Token(token_ata)` or something?

Reply:
The idea was to have a variant for each token in ```AssetWithDerivedAddress``` and so each token would have a match arm here. But I can see that all tokens would have the same code so we could just have one variant. Will update